### ISSUE TITLE
[a11y-defect-1]: Sitewide icon-only search button must have an accessible label to be understood by assistive technology

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-cms#20625]
### Point of contact

Trevor Pierce - wrote ticket on behalf of Martha Wilkes, OCTO Accessibility

### Severity level

1, Launchblocker. Critical. Must be fixed before launch.

### Details
The VA 508 office called out an issue with the sitewide search button as part of their `VA.GOV Application for Burial Benefits 21P-530EZ project` audit. When users tab to or focus the magnifying glass icon in the sitewide search dropdown it is announced as “Unlabeled 3 button, to activate press enter.”

---
![Image](https://github.com/user-attachments/assets/7ac637a3-c5d3-4d3a-b88a-40140a0677ee)

### Reproduction steps

1. Open the front page of production or staging VA.gov with [axe dev tools](https://deque.com/axe) installed on Chrome, Firefox, or Edge
2. Open developer tools before clicking "Search" to open the search dropdown
3. Run a scan using axe dev tools
4. Verify there is 1 critical issue on the page "Ensure buttons have discernible text"

### Proposed solution or next steps

[Line 809](https://github.com/department-of-veterans-affairs/vets-website/blob/b919258b52a022314b2570b22c5afce3eed6a82f/src/platform/site-wide/user-nav/components/SearchDropdownComponent.js#L809) of the `SearchDropdownComponent.js` conditionally shows text if the `buttonText` prop is passed to the component.

The VA accessibility specialists have determined when this prop is not passed (as is the case in the sitewide search menu) then the magnifying glass button is unlabeled, causing an error.

A good approach might be to conditionally show visible text or screen reader only text:

```javascript
{buttonText ? (
  <span className="button-text">{buttonText}</span>
) : (
  <span className="usa-sr-only">{buttonText}</span>
)}
```

It would also be a good idea to add a Cypress test with the following conditions:
1. The sitewide search dropdown is open
2. Test includes an `axeCheck` to prevent label regression

### References, articles, or WCAG support

1. [Understanding SC 4.1.2: Name, Role, Value (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value.html)

### Type of issue

- [x] Axe-core
- [x] Screenreader | assistive tech & device support
- [ ] Keyboard
- [ ] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Focus
- [ ] Headings
- [ ] Content style guide
- [x] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else

**Comment by FranECross** [comment]
@jilladams This bug is marked critical... Seems like this should be injected into the current sprint. Thoughts?

**Comment by jilladams** [comment]
Up to you and Michelle, but @Shamiaalam could maybe also help assign a criticality level. Options are a11y-defect-0 being most critical through a11y-defect-4. If you start to add the label to the ticket, there's a description of what they imply that might be useful: 

![Image](https://github.com/user-attachments/assets/ef9eb1ba-7dd0-4ea0-9ed5-c51162e22089)

**Comment by FranECross** [comment]
@jilladams Thanks! I like your suggestion - @Shamiaalam Will you please review this defect and then assign the criticality level label as suggested by Jill above? Thanks so much everyone!

**Comment by Shamiaalam** [comment]
> [@jilladams](https://github.com/jilladams) This bug is marked critical... Seems like this should be injected into the current sprint. Thoughts?

its a serious issue, is it possible to push this issue. into current sprint?

**Comment by FranECross** [comment]
@Shamiaalam Thanks so much for the quick review and it's criticality level. I'll inject into the sprint and will slack engineers to let them know. cc @jilladams

**Comment by Shamiaalam** [comment]
> [@Shamiaalam](https://github.com/Shamiaalam) Thanks so much for the quick review and it's criticality level. I'll inject into the sprint and will slack engineers to let them know. cc [@jilladams](https://github.com/jilladams)

thanks @FranECross

**Comment by chriskim2311** [comment]
@Shamiaalam Can you take a look at this when you get a chance and if all looks good can you close it out thanks!

**Comment by Shamiaalam** [comment]
@chriskim2311 it looks good! Approved ✅ Closing this.
